generator/src/generator.rs

```diff
@@ -822,3 +832,57 @@ impl Generate for RegisterCredentialsGenerator {
         Ok(tx)
     }
 }
+
+/// A generator that makes register data transactions with random data of a
+/// specified size.
+pub struct RegisterDataGenerator {
+    args:  CommonArgs,
+    nonce: Nonce,
+    rng:   StdRng,
+    size:  u16,
+}
+
+impl RegisterDataGenerator {
+    pub async fn instantiate(
+        mut client: v2::Client,
+        args: CommonArgs,
+        register_data_args: RegisterDataArgs,
+    ) -> anyhow::Result<Self> {
+        anyhow::ensure!(
+            register_data_args.size <= 256,
+            "Data size must be at most 256 bytes."
+        );
+
+        // Get the initial nonce.
+        let nonce = client
+            .get_next_account_sequence_number(&args.keys.address)
+            .await?;
+
+        let rng = StdRng::from_entropy();
+        Ok(Self {
+            args,
+            nonce: nonce.nonce,
+            rng,
+            size: register_data_args.size,
+        })
+    }
+}
+
+impl Generate for RegisterDataGenerator {
+    fn generate(&mut self) -> anyhow::Result<AccountTransaction<EncodedPayload>> {
+        let data_bytes: Vec<u8> = (0..self.size).map(|_| self.rng.gen()).collect();
+        let data: RegisteredData = data_bytes.try_into().unwrap();
```

Comment:
Not sure how much it matters, but do we want to panic here?
```suggestion
        let data: RegisteredData = data_bytes.try_into()?;
```

Reply:
I'll change it, but I don't think it should really matter. The size should be at most 256, so the `try_into` should never fail.